### PR TITLE
Task for multiplicity distribution in Pb-Pb

### PR DESCRIPTION
**Pull request by abmodak**
This is the task for analysing multiplicity distribution in Pb-Pb

### CODE REVIEW

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+///        it is meant to be a blank page for further developments.
+/// \author Abhi Modak (contact: abhi.modak@cern.ch)
+/// \help: To develop this code, I took help from the following codes and O2 analysis tutorial
+// 1. https://github.com/AliceO2Group/O2Physics/blob/master/PWGMM/Mult/Tasks/dndeta.cxx
+// 2. https://github.com/AliceO2Group/O2Physics/blob/master/PWGMM/Mult/Tasks/dndeta-hi.cxx
+// 3. https://github.com/AliceO2Group/O2Physics/blob/master/PWGLF/Mult/Tasks/puremc-dndeta.cxx
+// 4. O2 analysis tutorial: https://indico.cern.ch/event/1267433/
+
+#include <Math/Vector4D.h>
+#include <array>
```
[the remark this answers is not in the document]
Comment (abmodak) [reply PWGMM/Mult/Tasks/heavy-ion-mult.cxx]: I have removed most of the includes. Please let me know if I need to remove some more includes.

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+static constexpr TrackSelectionFlags::flagtype trackSelectionDCA =
+  TrackSelectionFlags::kDCAz | TrackSelectionFlags::kDCAxy;
+
+struct HeavyIonMultiplicity {
+
+  HistogramRegistry histos{"histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+  Service<O2DatabasePDG> pdg;
+  Preslice<TrackMCRecTable> perCollision = aod::track::collisionId;
+  
+  Configurable<bool> isMC{"isMC", true, "only for MC"};
```
[the remark this answers is not in the document]
Comment (abmodak) [reply PWGMM/Mult/Tasks/heavy-ion-mult.cxx]: Now, I have used doprocess... function.

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  Configurable<bool> isData{"isData", true, "only for Data"};
+
+  void init(InitContext const&){
+
+    const AxisSpec axisEvent{4, -0.5, 3.5, "#Event"};
```
[the remark this answers is not in the document]
Comment (abmodak) [reply PWGMM/Mult/Tasks/heavy-ion-mult.cxx]: Now, I declared the axis specification outside the init() function.

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+    if(RecCollisions.size() == 0 || RecCollisions.size() > 1) return;
+    
+    auto NchGenTracks = 0;
+    bool Inelgt0Gen = false;
+    auto NchRecTracks = 0;
+    bool Inelgt0Rec = false;
+      
+    for (auto& particle : GenParticles) {
+      if (!particle.isPhysicalPrimary()) continue;
```
[the remark this answers is not in the document]
Comment (abmodak) [reply PWGMM/Mult/Tasks/heavy-ion-mult.cxx]: Done

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+      if (!particle.producedByGenerator()) continue;
+      auto pdgParticle = pdg->GetParticle(particle.pdgCode());
```
[the remark this answers is not in the document]
Comment (abmodak) [reply PWGMM/Mult/Tasks/heavy-ion-mult.cxx:169]: Done

**Comment by abmodak** [comment]
> Please fix the formatting (you can run `clang-format` yourself or accept the PR that was automatically opened in you fork of O2Physics) and make other minor requested changes.

I have changed the code as per your suggestions.

**Comment by abmodak** [comment]
I am closing the pull request. There is something I could not solved.

**Review by aalkin — APPROVED** [review]

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+AxisSpec axisEvent{4, -0.5, 3.5, "#Event"};
+AxisSpec axisVtxZ{800, -20, 20, "Vertex Z"};
+AxisSpec axisDCA = {601, -3.01, 3.01};
+AxisSpec axisPT = {1000, -0.05, 49.95};
+AxisSpec axisMult{2000, -0.5, 1999.5, "Multiplicity"};
+AxisSpec axisEta{200, -5, 5, "#eta"};
+AxisSpec axisMCEvent_ambiguity{6, -0.5, 5.5, "reco collisions per true collision"};
+struct HeavyIonMultiplicity {
+  HistogramRegistry histos{"histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+  Service<O2DatabasePDG> pdg;
```
Comment (aalkin) [inline PWGMM/Mult/Tasks/heavy-ion-mult.cxx:74]: Please use `o2::framework::O2DatabasePDG` as there is an ambiguity here

**Review by aalkin — APPROVED** [review]

File: PWGMM/Mult/Tasks/heavy-ion-mult.cxx (hunk 6 above)
Comment (abmodak) [reply PWGMM/Mult/Tasks/heavy-ion-mult.cxx]: Hi Anton, Thanks a lot for pointing out this.
Now it is done.

**Pull request merged**